File: android/src/main/java/com/gachlab/capacitor/keepawake/KeepAwakePlugin.java
package com.gachlab.capacitor.keepawake;

import com.getcapacitor.JSObject;
import com.getcapacitor.Plugin;
import com.getcapacitor.PluginCall;
import com.getcapacitor.PluginMethod;
import com.getcapacitor.annotation.CapacitorPlugin;

@CapacitorPlugin(name = "KeepAwake")
public class KeepAwakePlugin extends Plugin {

    private final KeepAwake implementation = new KeepAwake();

    @PluginMethod
    public void dontAllowSleep(PluginCall call) {
        try {
            boolean isAllowedSleep = implementation.dontAllowSleep(getActivity());
            JSObject ret = new JSObject();
            ret.put("isAllowedSleep", isAllowedSleep);
            call.resolve(ret);
        } catch (Exception e) {
            call.reject("Failed to set keep awake", e);
        }
    }

    @PluginMethod
    public void allowSleep(PluginCall call) {
        try {
            boolean isAllowedSleep = implementation.allowSleep(getActivity());
            JSObject ret = new JSObject();
            ret.put("isAllowedSleep", isAllowedSleep);
            call.resolve(ret);
        } catch (Exception e) {
            call.reject("Failed to allow sleep", e);
        }
    }

    @PluginMethod
    public void isKeepAwake(PluginCall call) {
        try {
            boolean isKeepAwake = implementation.isKeepAwake(getActivity());
            JSObject ret = new JSObject();
            ret.put("isKeepAwake", isKeepAwake);
            call.resolve(ret);
        } catch (Exception e) {
            call.reject("Failed to check keep awake state", e);
        }
    }
}

File: ios/Sources/KeepAwakePlugin/KeepAwakePlugin.swift
import Foundation
import Capacitor

@objc(KeepAwakePlugin)
public class KeepAwakePlugin: CAPPlugin, CAPBridgedPlugin {
    public let identifier = "KeepAwakePlugin"
    public let jsName = "KeepAwake"
    public let pluginMethods: [CAPPluginMethod] = [
        CAPPluginMethod(name: "dontAllowSleep", returnType: CAPPluginReturnPromise),
        CAPPluginMethod(name: "allowSleep", returnType: CAPPluginReturnPromise),
        CAPPluginMethod(name: "isKeepAwake", returnType: CAPPluginReturnPromise)
    ]
    private let implementation = KeepAwake()

    @objc func dontAllowSleep(_ call: CAPPluginCall) {
        implementation.dontAllowSleep { isAllowedSleep in
            call.resolve(["isAllowedSleep": isAllowedSleep])
        }
    }

    @objc func allowSleep(_ call: CAPPluginCall) {
        implementation.allowSleep { isAllowedSleep in
            call.resolve(["isAllowedSleep": isAllowedSleep])
        }
    }

    @objc func isKeepAwake(_ call: CAPPluginCall) {
        implementation.isKeepAwake { isKeepAwake in
            call.resolve(["isKeepAwake": isKeepAwake])
        }
    }
}
